dandiapi/api/tasks/zarr.py

```diff
@@ -75,84 +139,77 @@ def apply_modification(
         return file_updater
 
 
-def yield_files(client, zarr: ZarrArchive):
-    """Get all objects in the bucket, through repeated object listing."""
-    continuation_token = None
-    common_options = {'Bucket': zarr.storage.bucket_name, 'Prefix': zarr.s3_path('')}
-    while True:
-        options = {**common_options}
-        if continuation_token is not None:
-            options['ContinuationToken'] = continuation_token
-
-        # Fetch
-        res = client.list_objects_v2(**options)
-
-        # Yield this batch of files
-        yield res.get('Contents', [])
-
-        # If all files fetched, end
-        if res['IsTruncated'] is False:
-            break
-
-        # Get next continuation token
-        continuation_token = res['NextContinuationToken']
-
-
-def get_client():
-    storage = get_storage()
-    if isinstance(storage, MinioStorage):
-        return boto3.client(
-            's3',
-            endpoint_url=storage.client._endpoint_url,
-            aws_access_key_id=storage.client._access_key,
-            aws_secret_access_key=storage.client._secret_key,
-            region_name='us-east-1',
-        )
-
-    return storage.connection.meta.client
-
-
 @shared_task
-@atomic
 def ingest_zarr_archive(
     zarr_id: str, no_checksum: bool = False, no_size: bool = False, no_count: bool = False
 ):
-    client = get_client()
-    zarr: ZarrArchive = ZarrArchive.objects.select_for_update().get(zarr_id=zarr_id)
-
-    # Reset before compute
-    if not no_size:
-        zarr.size = 0
-    if not no_count:
-        zarr.file_count = 0
-
-    # Instantiate updater and add files as they come in
-    updater = SessionZarrChecksumUpdater(zarr_archive=zarr)
-    for files in yield_files(client, zarr):
-        # Update size and file count
-        if not no_size:
-            zarr.size += sum((file['Size'] for file in files))
-        if not no_count:
-            zarr.file_count += len(files)
-
-        # Update checksums
-        if not no_checksum:
-            updater.update_file_checksums(
-                [
-                    ZarrChecksum(
-                        md5=file['ETag'].strip('"'),
-                        path=file['Key'].replace(zarr.s3_path(''), ''),
-                    )
-                    for file in files
-                ]
+    # Set zarr status before ingest
+    with transaction.atomic():
+        try:
+            zarr: ZarrArchive = ZarrArchive.objects.select_for_update(nowait=True).get(
+                zarr_id=zarr_id
             )
+            zarr.status = ZarrArchive.Status.INGESTING
```

Comment:
I think it's worth checking if `zarr.status == ZarrArchive.Status.PENDING` before proceeding